src/webgpu/api/validation/resource_usages/textureUsageInPassEncoder.spec.ts

```diff
@@ -124,33 +127,74 @@ class TextureUsageTracking extends ValidationTest {
     });
   }
 
-  testValidationScope(): {
+  testValidationScope(
+    compute: boolean
+  ): {
     bindGroup0: GPUBindGroup;
     bindGroup1: GPUBindGroup;
     encoder: GPUCommandEncoder;
-    pass: GPURenderPassEncoder;
-    pipeline: GPURenderPipeline;
+    renderPass: GPURenderPassEncoder;
+    computePass: GPUComputePassEncoder;
+    renderPipeline: GPURenderPipeline;
+    computePipeline: GPUComputePipeline;
   } {
     // Create two bind groups. Resource usages conflict between these two bind groups. But resource
     // usage inside each bind group doesn't conflict.
     const view = this.createTexture({
       usage: GPUTextureUsage.STORAGE | GPUTextureUsage.SAMPLED,
     }).createView();
-    const bindGroup0 = this.createBindGroup(0, view, 'sampled-texture', '2d', undefined);
+    const bindGroup0 = this.createBindGroup(0, view, 'sampled-texture', '2d', undefined, compute);
     const bindGroup1 = this.createBindGroup(
       0,
       view,
       'writeonly-storage-texture',
       '2d',
-      'rgba8unorm'
+      'rgba8unorm',
+      compute
     );
 
     const encoder = this.device.createCommandEncoder();
-    const pass = this.beginSimpleRenderPass(encoder, this.createTexture().createView());
+    const renderPass = this.beginSimpleRenderPass(encoder, this.createTexture().createView());
+    const computePass = encoder.beginComputePass();
```

Comment:
You can't beginComputePass and then beginRenderPass on the same encoder; both require that the encoder be in a non-pass state beforehand.

Reply:
Oops. Looks like we can't share code between render pass and compute pass. Code snippet outside pass, say constructing bind groups, can be shared. I should have run the tests in Chromium.

Reply:
I think code can be shared between render and compute pass. You have another case where you do `const pass = compute ? beginComputePass(...) : beginSimpleRenderPass(...)` and that's fine.

Reply:
Agreed! We can use unsafe type cast as you suggested below to share code between render pass and compute pass. Done.